src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -764,24 +767,63 @@ public override void GenerateClassSpecifier(Class @class)
                 Write(" : {0}", string.Join(", ", bases));
         }
 
-        private void GenerateClassInternalsField(LayoutField field, Class @class)
+        private bool CanUseSequentialLayout(Class @class)
         {
-            TypePrinterResult retType = TypePrinter.VisitFieldDecl(
-                new Field { Name = field.Name, QualifiedType = field.QualifiedType });
+            if (@class.IsUnion)
+                return false;
 
-            PushBlock(BlockKind.Field);
+            var fields = @class.Layout.Fields;
 
-            if (!Options.GenerateSequentialLayout || @class.IsUnion)
-                WriteLine($"[FieldOffset({field.Offset})]");
-            Write($"internal {retType}");
-            if (field.Expression != null)
-            {
-                var fieldValuePrinted = field.Expression.CSharpValue(ExpressionPrinter);
-                Write($" = {fieldValuePrinted}");
+            if (fields.Count > 1)
+            { 
+                for (var i = 1; i < fields.Count; ++i)
+                {
+                    if (fields[i].Offset == fields[i - 1].Offset)
```

Comment:
Why is this necessary? Unless the type is a union which is already checked, fields at the same offset are impossible. If our AST reports this in any case, then it has a bug.

Reply:
> fields at the same offset are impossible. 

```
class Time
{
public:
    // a & b will share the same offset
    union 
    {
        int a;
        double b;
    };
    int test;
};```

Reply:
Indeed they are – we just inline fields of empty types. Perhaps we need to reconsider this.